### DISCUSSION TITLE
Capture at Power On with an Optional Escape

### DISCUSSION THREAD

**Discussion by pairahoos** [gopro/labs#857] [Q&A]
I've succeeded in creating a script which starts capture automatically on power on, thanks to the dashcam examples from this forum. However, I'm trying to add to the script to allow an escape control at startup. I don't always want this to be the default behavior, and it currently causes a loop that can't be escaped without pulling the SD card and resetting.

I tried adding a shutter press count with a "exit program", but this does not work. Not sure what I'm missing here.

Also, the "USB Powered" status is not an accurate label. The automatic capture is triggered by inserting the battery - without the USB.

Working with a Hero 10 Black, and here are the Blocky build, and QR. 

![Record at Power On Blocky](https://github.com/gopro/labs/assets/166753469/64481a54-22aa-4fd6-b78e-71c11a6eb30b)

![Record at Power On](https://github.com/gopro/labs/assets/166753469/50678755-a3bf-4b0f-aec7-be97c036faaf)

**Comment by dnewman-gpsw** [comment]
> Also, the "USB Powered" status is not an accurate label. The automatic capture is triggered by inserting the battery - without the USB.

I disagree with your finding here.  While inserting the battery is triggering a wake event under WAKE=2 (WAKE On Power, not Wake on USB), it will not fail 
![image](https://github.com/gopro/labs/assets/10856084/f08cadec-9016-45e1-987f-008f182c206d)

which is only testing for USB power.

The problem with your code is the sleep 10 seconds.  Any sleep over 9 seconds will shutdown the camera, during which time no button presses will be counted.   

This is an improved version. With a second loop to monitor for shutter presses.  It uses !Z3 to mute the buttons from their typical behaviors.

![Record On Power + Button Cancel-4_12_2024, 08_45_02](https://github.com/gopro/labs/assets/10856084/88e4b8a8-6ab0-4358-9a76-d59f4d825a5b)

**Reply by pairahoos** [reply]
David - Thanks for the quick response. Interesting about the battery vs. USB power. That's not what I've been experiencing. But I've cleared my SD card, done a Boot Commands Reset and a Factory Reset. So hopefully I'm starting with a clean slate. I also did not get the shurdown after a 9 second pause, so that bit of code may never have been reached.

I'm looking forward to looking at your improved version, but when I click on the above I get "This private-user-images.githubusercontent.com page can’t be found"

Thanks,
John

**Reply by pairahoos** [reply]
David - the link in the email worked, and I'm able to see the design in Blocky. Need to learn about "added QR command"!

**Reply by dnewman-gpsw** [reply]
In Blockly not every weird command was implemented, so for the extras like !Z3 I used the "added QR command".

**Reply by pairahoos** [reply]
Cleared my SD card, did a Reset Boot Commands, did a Factory reset, then loaded your improved version.

Without a battery, applied USB power. Camera immediately began recording, no "shutter presses to exit" prompt. 

Pulled the USB power. Reinstalled the battery - without USB power - and experienced exactly the same behavior. Immediate recording launched, no prompt for "shutter presses to exit."

**Reply by pairahoos** [reply]
Does Loop 2 need a pause? Wondering if its getting the 10 loops almost instantaneously, without even enough time to display the prompt?

**Reply by dnewman-gpsw** [reply]
What firmware version are you running?

**Reply by dnewman-gpsw** [reply]
This requires 1.60.70 firmware.  https://bit.ly/LABS_H10_1_60_70

**Reply by pairahoos** [reply]
Progress. I have the 10_1_60_70 installed. The Record at Power On + Button Cancel worked the first time. But on the second and subsequent power on's, there is no prompt or opportunity to interrupt the automatic record at power on. 

I did add a 1 second pause in the loop, and limited it to 9.


![Record at Power On + Button Cancel Blocky](https://github.com/gopro/labs/assets/166753469/2dcbbf9d-1fc4-44f0-a1b1-4cdcd60d8c15)

**Reply by dnewman-gpsw** [reply]
That shouldn't have been necessary as the print message take a second, and now it will flash.

**Reply by pairahoos** [reply]
I'm not sure this revised program is controlling things. I've only seen the "shutter" prompt intermittently.

I just cleared the SD card, copied the UPDATE file to the card, and did a !RESET and a !FRESET after the camera successfully completed the update. The camera behaves normally at this point. But when I pull the power, and re-supply the power - without any additional programming - the camera immediately begins capture. There is no prompt for "shutter" or a prompt for "record w USB Pwr." 

It feels as if an earlier version of the "Record w USB Pwr" is still in control, one without the escape logic.

What am I missing in getting the camera back to a clean slate / factory defaults?

**Reply by dnewman-gpsw** [reply]
```!RESET``` must say "approved" after the button presses, otherwise it does nothing. If you just want the boot command to stop just do ```!MBOOT=""```

**Reply by dnewman-gpsw** [reply]
There is no other Labs mechanism to change capture behavior without BOOT extensions.

**Reply by pairahoos** [reply]
Wow! I'm am stuck in a loop. I cleared the SD card, and added the Update file. After a successful update, I pointed at the !FRESET GoProQR, and "approved" it with the three button presses. After the factory reset I pointed at the !MBOOT="" QR code and got a beep and screen display acknowledging it. I pulled the power, and powered off. When I powered on with battery alone, the camera immediately began capture at power on. No opportunity to interrupt! Same behavior with just USB power: immediate Capture at power on; no opportunity to interrupt. Also, when I press the "Shutter" button - the camera powers down! It fires right back up and begins a new capture. Quite the loop - and behavior I occasionally would appreciate - but not every time I provide power.

**Reply by dnewman-gpsw** [reply]
Use https://gopro.github.io/labs/control/metadata/ to share the metadata from a result (automatic) capture.  This might report any odd settings.

Also share MISC/qrlog.txt

**Reply by pairahoos** [reply]
Here is the metadata from one of the MP4 files automatically captured:
DEVC 
  DVID 1
  DVNM Global Settings
  VERS 8, 2, 2
  FMWR H21.01.01.50.70
  LINF LKO1092103402079
  CINF 43, -88, 124, 13, 61, 11, 74, -51, -73, 92, -55, 26, -73, -18, 27, -111
  CASN C3464250079120
  MINF HERO10 Black
  MUID 226273323, -850785475, 449404087, -1860440393, 0, 0, 0, 0
  CMOD 12
  MTYP 0
  OREN U
  DZOM Y
  DZST 0
  SMTR N
  PRTN Y
  PTWB AUTO
  PTSH MED
  PTCL NATURAL
  EXPT AUTO
  PIMX 1600
  PIMN 100
  PTEV 0.0
  RATE .empty.
  VFOV W
  SROT 16.0110
  ZFOV 123.6363
  EISE Y
  EISA HS High
  HLVL N
  AUPT N
  AUDO AUTO
  BROD .empty.
  BRID 0
  PVUL F
  PRJT GPRO
  SOFF 0
  CLKS 1
  CDAT 0x00000000661EA500
  SCTM 0
  PRNA 1
  PRNU 0
  SCAP N
  CDTM 0
  DUST NO_LIMIT
  VRES 3840, 2160
  VFPS 30000, 1001
  HSGT OFF
  BITR STANDARD
  MMOD STEREO
  RAMP .empty.
  TZON 32767
  CLKC 0
  DZMX 1.4000
  PWPR PERFORMANCE
  ORDP Y
  CLDP Y
DEVC 
  DVID HLMT
  DVNM Highlights

Here is the qrlog:

[2024-04-15 15:42:03] Labs:Main Start 
[2024-04-15 15:42:02] Labs:Main Start 
[2024-04-15 15:42:02] Labs:Initialize 
[2024-04-15 15:42:02] Read QR:!FRESET 
[2021-01-01 12:00:09] Labs:Main Start 
[2021-01-01 12:00:09] Labs:Initialize 
[2021-01-01 12:00:09] Read QR:!MBOOT="" 
[2024-04-16 16:16:18] Read QR:!MBOOT="" 
[2024-04-16 16:19:14] Labs:Main Start 
[2024-04-16 16:19:14] Labs:Initialize 
[2024-04-16 16:22:46] Labs:Main Start 
[2024-04-16 16:22:46] Labs:Initialize 
[2024-04-16 16:24:05] Labs:Main Start 
[2024-04-16 16:24:05] Labs:Initialize 
[2024-04-16 16:24:21] Labs:Main Start 
[2024-04-16 16:24:21] Labs:Initialize

**Reply by dnewman-gpsw** [reply]
This is not the 1.60.70 firmware, required for some the new scripting (this is from 2022.) The metadata dump is missing all the User Metadata which contains all the Labs settings, so there isn't enough here to learn what could be happening.

**Comment by pairahoos** [comment]
I redownloaded the Update file from the link you provided: https://bit.ly/LABS_H10_1_60_70. I successfully updated the camera, and did a factory reset. I'm still getting the automatic capture at power on! I looked at the meta data - and its showing FMWR | H21.01.01.50.70.
 
I went looking for 1.60.70 firmware, and ended up here: https://gopro.github.io/labs/ . It shows Hero10 Black v1.50.70, December 2022. I don't see 1.60.70 anywhere.

![image](https://github.com/gopro/labs/assets/166753469/4e0d0291-a827-44c3-8cb1-c8dfa0f70d22)

**Reply by dnewman-gpsw** [reply]
That is why I gave the link to 1.60.70.  The version was temporally removed for the default download, as it doesn't work in all languages.  

The experience you describe is not a known Labs issue.  Just install stock firmware and if the problem persist contact support. https://gopro.com/en/us/support/hero10-black-product-update/windows
